### PR TITLE
Utkarsha/migration banner for dark mode

### PR DESCRIPTION
**Pull request by utkarsha-deriv**
## Changes:

Please provide a summary of the change.

### Screenshots:

Please provide some screenshots of the change.

### CODE REVIEW

File: packages/appstore/src/components/appstore-banner/appstore-banner-image.tsx [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import SVGMigrateDesktopImage from 'Assets/svgs/banner/svg-migrate-desktop.svg';
+import SVGMigrateDesktopImageDark from 'Assets/svgs/banner/svg-migrate-desktop-dark.svg';
 import SVGMigrateMobileImage from 'Assets/svgs/banner/svg-migrate-mobile.svg';
+import SVGMigrateMobileImageDark from 'Assets/svgs/banner/svg-migrate-mobile-dark.svg';
 
 export type BannerImageProps<T> = {
```
Comment (shaheer-deriv) [inline packages/appstore/src/components/appstore-banner/appstore-banner-image.tsx]: TBannerImageProps

File: packages/appstore/src/components/appstore-banner/appstore-banner-image.tsx [hunk 2]
```diff
@@ -13,6 +15,8 @@ type TAppstoreBannerImageList = keyof typeof AppstoreBannerImageList;
 const AppstoreBannerImageList = {
     svg_migrate_desktop: SVGMigrateDesktopImage,
     svg_migrate_mobile: SVGMigrateMobileImage,
+    svg_migrate_desktop_dark: SVGMigrateDesktopImageDark,
+    svg_migrate_mobile_dark: SVGMigrateMobileImageDark,
 };
 
 const AppstoreBannerImage = ({ image, className, width }: BannerImageProps<TAppstoreBannerImageList>) => {
```
Comment (shaheer-deriv) [inline packages/appstore/src/components/appstore-banner/appstore-banner-image.tsx]: TAppstoreBannerImageListProps

File: packages/appstore/src/components/appstore-banner/svg-migration-banner.tsx [hunk 3]
```diff
@@ -66,7 +67,11 @@ const SVGMigrationBanner = observer(() => {
                     <Localize i18n_default_text='Upgrade now' />
                 </Button>
             </div>
-            <AppstoreBannerImage image={image} className='appstore-banner__image' />
+            {is_dark_mode_on ? (
+                <AppstoreBannerImage image={image_dark_mode} className='appstore-banner__image' />
+            ) : (
+                <AppstoreBannerImage image={image} className='appstore-banner__image' />
+            )}
```
Comment (shaheer-deriv) [inline packages/appstore/src/components/appstore-banner/svg-migration-banner.tsx]: <AppstoreBannerImage image={`is_dark_mode_on? image_dark_mode:image`} className='appstore-banner__image' />
will this work?

Comment (utkarsha-deriv) [reply packages/appstore/src/components/appstore-banner/svg-migration-banner.tsx]: yess

**Review by shaheer-deriv — APPROVED** [review]

File: packages/appstore/src/components/appstore-banner/appstore-banner-image.tsx [hunk 4]
```diff
@@ -2,20 +2,20 @@ import React from 'react';
...
+type TAppstoreBannerImageListProps = keyof typeof AppstoreBannerImageList;
 
 const AppstoreBannerImageList = {
     svg_migrate_desktop: SVGMigrateDesktopImage,
     svg_migrate_mobile: SVGMigrateMobileImage,
 };
 
-const AppstoreBannerImage = ({ image, className, width }: BannerImageProps<TAppstoreBannerImageList>) => {
+const AppstoreBannerImage = ({ image, className, width }: TBannerImageProps<TAppstoreBannerImageListProps>) => {
     const Component = AppstoreBannerImageList[image] as React.ElementType;
```
Comment (shaheer-deriv) [inline packages/appstore/src/components/appstore-banner/appstore-banner-image.tsx]: does typescript throw error if `as React.ElementType` is omitted?

File: packages/appstore/src/components/appstore-banner/appstore-banner-image.tsx [hunk 5]
```diff
@@ -2,20 +2,20 @@ import React from 'react';
...
-type TAppstoreBannerImageList = keyof typeof AppstoreBannerImageList;
+type TAppstoreBannerImageListProps = keyof typeof AppstoreBannerImageList;
 
 const AppstoreBannerImageList = {
     svg_migrate_desktop: SVGMigrateDesktopImage,
     svg_migrate_mobile: SVGMigrateMobileImage,
 };
 
-const AppstoreBannerImage = ({ image, className, width }: BannerImageProps<TAppstoreBannerImageList>) => {
+const AppstoreBannerImage = ({ image, className, width }: TBannerImageProps<TAppstoreBannerImageListProps>) => {
```
Comment (shaheer-deriv) [inline packages/appstore/src/components/appstore-banner/appstore-banner-image.tsx]: rename prop to `class_name`?

File: packages/appstore/src/components/appstore-banner/appstore-banner.scss [hunk 6]
```diff
@@ -66,5 +69,20 @@
                 padding-inline: 1.6rem;
             }
         }
+
+        &-overlay {
+            position: absolute;
+            z-index: 1;
+            width: 622px;
```
Comment (shaheer-deriv) [inline packages/appstore/src/components/appstore-banner/appstore-banner.scss]: can these `px` values be `rem`?

File: packages/appstore/src/components/appstore-banner/appstore-banner.scss [hunk 7]
```diff
@@ -47,14 +51,13 @@
             align-items: baseline;
 
             @include desktop {
-                background: var(--mt5-migration-banner);
-                &__text {
-                    width: 50%;
-                }
+                padding-left: 48px;
```
Comment (amina-deriv) [inline packages/appstore/src/components/appstore-banner/appstore-banner.scss]: rem?

File: packages/appstore/src/components/appstore-banner/appstore-banner-image.tsx (hunk 4 above)
Comment (utkarsha-deriv) [reply packages/appstore/src/components/appstore-banner/appstore-banner-image.tsx]: works fine

**Review by amina-deriv — APPROVED** [review]

**Pull request merged**